ipa-core/src/protocol/context/mod.rs

```diff
@@ -822,15 +867,42 @@ mod tests {
 
     /// Ensure global record order across shards is consistent.
```

Comment:
the comment needs to be changed, otherwise lgtm

Reply:
heh, I can't approve my own PRs

Reply:
I thought the purpose of this test is to ensure we can use stream API, ensuring global record order in two tests seems redundant. But that's a nit, lets merge it and follow up if needed

Reply:
ahh I see you're point. I can open a small PR to clarify that. we want global ordering, and we want that to work with both APIs